src/Symfony/Bridge/Doctrine/Validator/Constraints/UniqueEntityValidator.php

```diff
@@ -287,7 +287,9 @@ private function getFieldValues(mixed $object, ClassMetadata $class, array $fiel
                 throw new ConstraintDefinitionException(sprintf('The field "%s" is not a property of class "%s".', $fieldName, $objectClass));
             }
 
-            $fieldValues[$entityFieldName] = $this->getPropertyValue($objectClass, $fieldName, $object);
+            $fieldValues[$entityFieldName] = $isValueEntity && $object instanceof ($class->getName())
```

Comment:
Hm could we potentially have the same problem in case `$isValueEntity` is true but `$object instanceof ($class->getName())` is false? 

When is that the case actually :thinking: Is it if I just pass a different `entityClass` on the constraint?

Reply:
According to the tests, when that happens an exception is thrown, see:

https://github.com/symfony/symfony/blob/99f279b3383efd4f88ac79f1ad0572cf7b9bf1de/src/Symfony/Bridge/Doctrine/Validator/Constraints/UniqueEntityValidator.php#L133-L139

Reply:
Ok got it :+1: